src/history/hs_rec.c

```diff
@@ -690,10 +690,18 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
 
             /*
              * Calculate reverse modify and clear the history store records with timestamps when
-             * inserting the first update.
+             * inserting the first update. Always write on-disk data store updates to the history
+             * store as a full update, because the on-disk update will be the base update for all
+             * the updates are previous to the on-disk update.
```

Comment:
updates are previous to the on-disk update. -> older than the previous on-disk update.

Reply:
Fixed.